### PR TITLE
Cm/carnival activity

### PR DESCRIPTION
**Pull request by CheezItMan**
## Description

I created a version of the carnival activity for C13.  It's a little different in a couple of ways:

1.  We create Zoom breakout rooms to form our groups.  On the bright side things are totally random.
1.  We will use branches an PRs.  This is because:
    - Students are further along now
    - This will more closely resemble how they might work on bEtsy using Zoom.

I also created an updated video documenting this.  For doing the activity I can do this with one instructor working with me to demonstrate the process.

## Feedback Requested
Please utilize the built-in GitHub Reviewers functionality to assign necessary reviewers. Use this section to further specify portions of the PR that you would like additional or focused feedback on.

### CODE REVIEW

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+1. **Split into groups**: The class will be divided into groups using the breakout sections feature of Zoom, these assignments are random.
+   1. Breakout Room 1:  Aardvark
+   2. Breakout Room 2:  Tarsier
+   3. Breakout Room 2:  Gecko
+   4. Breakout Room 2:  Sloth
+   5. Breakout Room 2:  Lionfish
+   6. Breakout Room 2:  Pangolin
+   7. Breakout Room 2:  Rhino
+   8. Breakout Room 2:  Curassow
+2. **Fork the activity repo**: One person in your group should fork the carnival-recipes repo and then add everyone else as a collaborator.
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:30]: I don't see a link to the repo anywhere. Should we add it here?

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+### Building the recipe
+
+Once everyone in your group has completed the baseline setup instructions above you're ready to start building the recipe together.
+
+#### General advice
+
+There are a few things to look out for when working through this activity:
+
+- If you're still unsure about what a merge conflict is, or how to resolve one please review the [Merge Conflicts curriculum](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/00-programming-fundamentals/git-merge-conflicts.md).
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:45]: I'm trying to remember if this is a reading we've already assigned them. If not, we should probably assign it to them the day before this activity.
Even if we have, it might be worth putting a thing in the hw the day before about reviewing this reading.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+- You can review the [slides](https://docs.google.com/presentation/d/15OFyyoNJevzk7IpdZCP6VwMLju0MVS2pv0qeETLOa6Q/edit?usp=sharing) and [example walkthrough video](https://adaacademy.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=ec7dbf60-e59b-4317-a4a5-abbe002eb7ea) if you are unsure about how to complete the activity.
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:46]: I appreciate you making an example video as well. I'm a little confused about how we expect they'd utilize this.
I wonder if we should either (1) assign this as a video to watch before the activity or (2) suggest that if they get stuck, one of the students do a screen share so that they can all watch the video together? But also the intention is that 2 or 3 instructors will be popping between the breakout rooms to support, right? If so, that feels to me like probably a more effective way to clarify things than asking the whole group to stop and watch a video...so I'm less sold on option 2. 
And also not sure that option 1 is necessary either. 
But curious for other thoughts

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+#### Recipe building process
+
+Because each member of your group now has a different version of the recipe file in their local repository, building the complete recipe by merging those versions together will result in merge conflicts. To handle those merge conflicts sensibly your group should use the following process to construct the final recipe file:
+
+1. Split your group into two teams of two or three.
+2. Each person on a team will create a branch with `git checkout -b BRANCH-NAME`, selecting their own branch name based off the recipe scrap they have, and commit the changes they have made to the recipe.
+3. Each team member will push their changes up to github with `git push origin BRANCH-NAME`.
+4. Both teams then will use the _round robin_ strategy separately.
+    - Each team member will open a **pull request** trying to merge their branch on github into `master`.  **Be very careful to make the PR against your repository and NOT ada Gold.**
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:61]:
```suggestion
    - Each team member will open a **pull request** trying to merge their branch on github into `master`.  **Be very careful to make the PR against your forked repository and NOT AdaGold.**
```

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+- If you find that the result of one of your merges is missing critical ingredients or any numbered steps, you should go back and double check that you didn't lose any lines while resolving a merge conflict.
+- Your recipe scraps may have duplicate lines. We've added these in to more accurately portray the full range of potential merge conflicts. **Important**: you should get rid of any duplicate lines while merging.
+- There's no "right" way to do this activity, all of the strategies listed below are valid ways to build the recipe file.
+- Have fun!
+
+#### Recipe building process
+
+Because each member of your group now has a different version of the recipe file in their local repository, building the complete recipe by merging those versions together will result in merge conflicts. To handle those merge conflicts sensibly your group should use the following process to construct the final recipe file:
+
+1. Split your group into two teams of two or three.
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:57]: Just thought the original wording made it sound like the students needed to make a choice about group size, which felt a little confusing to me.
```suggestion
1. Split your group into two teams of three. (There may be a group of two if your group only has 5 people.)
```

Comment (jmaddox19) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:57]: This does make me realize we may need a specific strategy for what to do about the one group of 5 that will exist.
Also makes me realize it will be especially helpful with this activity if we know ahead of time whether anyone will be absent.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+## Instructions
+
+### Baseline Setup
+
+Before we can start putting the recipe together we need to make sure that everyone in your group is working from the same starting point and can effectively contribute to the recipe building effort.
+
+1. **Split into groups**: The class will be divided into groups using the breakout sections feature of Zoom, these assignments are random.
+   1. Breakout Room 1:  Aardvark
+   2. Breakout Room 2:  Tarsier
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:23]: I know some people have strong opinions about using 1s for all numbered lists. But I assume this was an intentional choice for some particular reason.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+- You can review the [slides](https://docs.google.com/presentation/d/15OFyyoNJevzk7IpdZCP6VwMLju0MVS2pv0qeETLOa6Q/edit?usp=sharing) and [example walkthrough video](https://adaacademy.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=ec7dbf60-e59b-4317-a4a5-abbe002eb7ea) if you are unsure about how to complete the activity.
+- The end result should be a real, usable recipe. The original recipe files were all complete works from various recipe websites online.
+- If you find that the result of one of your merges is missing critical ingredients or any numbered steps, you should go back and double check that you didn't lose any lines while resolving a merge conflict.
+- Your recipe scraps may have duplicate lines. We've added these in to more accurately portray the full range of potential merge conflicts. **Important**: you should get rid of any duplicate lines while merging.
+- There's no "right" way to do this activity, all of the strategies listed below are valid ways to build the recipe file.
+- Have fun!
+
+#### Recipe building process
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:53]: To be honest, I find the nested list of lists of lists disorienting and hard to follow. 
I am inclined to suggest we just reuse the instructions from the [original README](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/00-programming-fundamentals/exercises/merge-conflict.md) and count on the instructors' support to give further clarity.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+### Baseline Setup
+
+Before we can start putting the recipe together we need to make sure that everyone in your group is working from the same starting point and can effectively contribute to the recipe building effort.
+
+1. **Split into groups**: The class will be divided into groups using the breakout sections feature of Zoom, these assignments are random.
+   1. Breakout Room 1:  Aardvark
+   2. Breakout Room 2:  Tarsier
+   3. Breakout Room 2:  Gecko
+   4. Breakout Room 2:  Sloth
```
[the remark this answers is not in the document]
Comment (CheezItMan) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:25]: I'd expected that we would hold a Zoom call with the entire cohort and we'll run breakout rooms to break the groups up into groups that way.  We're not doing separate zoom calls per roundtable.  

We could also do separate roundtables with each group.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md (hunk 1 above)
Comment (CheezItMan) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:30]: Since we use could use these instructions next cohort is the worst occurs, then I figured we would put the link in the calendar.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md (hunk 2 above)
Comment (CheezItMan) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:45]: It's in the calendar

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md (hunk 3 above)
Comment (CheezItMan) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:46]: If they get confused they can refer to the video, I'm not sure it's needed, but included it to parallel the original assignment which included it.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md (hunk 7 above)
Comment (CheezItMan) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:53]: The original doesn't use branches.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 9]
```diff
@@ -19,18 +19,18 @@ With this activity you will be able to encounter and resolve merge conflicts in
...
-4. **Find your recipe scrap**: Each group is assigned six recipe scraps that should all be part of the same recipe. You can split these scraps up amongst your group however you wish. The scraps assigned to each group are listed in the carnival-recipes repo. If your group has fewer than six people someone may need to take multiple scraps.
-5. **Add your recipe scrap**: Copy the contents of your recipe scrap into the recipe file and then add and commit your changes to the recipe file. After completing this step each person in your group should have **different** content within the recipe file in their local repository.
+   - Breakout Room 1:  Aardvark
+   - Breakout Room 2:  Tarsier
+   - Breakout Room 2:  Gecko
+   - Breakout Room 2:  Sloth
+   - Breakout Room 2:  Lionfish
+   - Breakout Room 2:  Pangolin
+   - Breakout Room 2:  Rhino
+   - Breakout Room 2:  Curassow
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:29]: Is this what you intended here?
```suggestion
   - Breakout Room 3:  Gecko
   - Breakout Room 4:  Sloth
   - Breakout Room 5:  Lionfish
   - Breakout Room 6:  Pangolin
   - Breakout Room 7:  Rhino
   - Breakout Room 8:  Curassow
```

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md (hunk 3 above)
Comment (jmaddox19) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:46]: I'm just hesitant to include something like a video in this group context because I fear it could cause confusion. ("Are we all supposed to watch this right now? Have we seen this video before? Should we all watch it separately and come back together?")
Would you be up for just taking it out and counting on the rotating instructors to provide guidance instead of the video?

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md (hunk 2 above)
Comment (jmaddox19) [reply 00-programming-fundamentals/exercises/merge-conflict-remote.md:45]: Oh, ok.

File: 00-programming-fundamentals/exercises/merge-conflict-remote.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+1. Split your group into two teams of three. (There may be a group of two if your group only has 5 people.)
+2. Each person on a team will create a branch with `git checkout -b BRANCH-NAME`, selecting their own branch name based off the recipe scrap they have, and commit the changes they have made to the recipe.
+3. Each team member will push their changes up to github with `git push origin BRANCH-NAME`.
+4. Both teams then will use the _round robin_ strategy separately.
+    - Each team member will open a **pull request** trying to merge their branch on github into `master`.  **Be very careful to make the PR against your forked repository and NOT AdaGold.**
+    - If there is a report of a merge conflict they will need to:
+     - Pull the current state of master into their branch with `git pull origin master`
+     - Resolve the merge conflicts and commit the result
+     - Push the result up to git hub with `git push origin BRANCH-NAME`
+     - Then attempt to merge their pull request.  If new changes have happened on master they may have to repeat step 1 above.
```
Comment (jmaddox19) [inline 00-programming-fundamentals/exercises/merge-conflict-remote.md:66]: It reads like these should be nested within the above bullet. I'm not sure if adding 2 spaces will fix it or not.
```suggestion
       - Pull the current state of master into their branch with `git pull origin master`
       - Resolve the merge conflicts and commit the result
       - Push the result up to git hub with `git push origin BRANCH-NAME`
       - Then attempt to merge their pull request.  If new changes have happened on master they may have to repeat step 1 above. 
```
